google/cloud/aiplatform/models.py

```diff
@@ -110,12 +110,50 @@ def __init__(
             credentials=credentials,
             resource_name=endpoint_name,
         )
-        self._gca_resource = self._get_gca_resource(resource_name=endpoint_name)
+
+        endpoint_name = utils.full_resource_name(
+            resource_name=endpoint_name,
+            resource_noun="endpoints",
+            project=project,
+            location=location,
+        )
+
+        # Lazy load the Endpoint gca_resource until needed
+        self._gca_resource = gca_endpoint_compat.Endpoint(name=endpoint_name)
 
         self._prediction_client = self._instantiate_prediction_client(
             location=self.location, credentials=credentials,
         )
 
+    def _skipped_getter_call(self) -> bool:
+        """Check if GAPIC resource was populated by call to get/list API methods
+
+        Returns False if `_gca_resource` is None or fully populated. Returns True
+        if `_gca_resource` is partially populated
+        """
+        return self._gca_resource and not self._gca_resource.create_time
+
+    def _sync_gca_resource_if_skipped(self) -> None:
+        """Sync GAPIC service representation of Endpoint class resource only if
+        get_endpoint() was never called."""
+        if self._skipped_getter_call():
+            self._gca_resource = self._get_gca_resource(
+                resource_name=self._gca_resource.name
+            )
+
+    def _sync_gca_resource(self) -> None:
+        """Sync GAPIC service representation of Endpoint resource once."""
+        if self._skipped_getter_call():
+            self._sync_gca_resource_if_skipped()
+        else:
+            super()._sync_gca_resource()
+
+    def _assert_gca_resource_is_available(self) -> None:
+        """Ensures Endpoint getter was called at least once before
+        asserting on gca_resource's availability."""
+        self._sync_gca_resource_if_skipped()
+        super()._assert_gca_resource_is_available()
```

Comment:
It seems like the implementation can be reduced with the same control flow if

1.  The above lines (154, 155) are switched. ie: Assert the resource is available and then sync if getter was skiiped.

2. Remove the overridden `_sync_gca_resource` method.